ansible/README.md

```diff
@@ -64,47 +67,73 @@ This needs to be done only once per environment. It will install necessary prere
-ansible-playbook -i environments/<environment> couchdb.yml
-ansible-playbook -i environments/<environment> couchdb.yml -e mode=initdb
 cd <openwhisk_home>
 gradle distDocker
 cd ansible
+ansible-playbook -i environments/<environment> couchdb.yml
+ansible-playbook -i environments/<environment> couchdb.yml -e mode=initdb
 ansible-playbook -i environments/<environment> openwhisk.yml
 ```
 
 You need to run "initdb" on couch **every time** you deploy couchdb to initialize the db.
 
 ### Deploying Using Cloudant
-- Enter your cloudant credentials in `environments/<environment>/group_vars/all`
-- Comment out the couchdb section
+- Make sure the cloudant section is **uncommented** in in `environments/<environment>/group_vars/all`
+- Make sure the couchdb section is **commented out**
+- Enter your cloudant credentials in the cloudant section
 - Then execute
 
 ```
-ansible-playbook -i environments/<environment> cloudant.yml -e mode=initdb
 cd <openwhisk_home>
 gradle distDocker
 cd ansible
+ansible-playbook -i environments/<environment> cloudant.yml -e mode=initdb
 ansible-playbook -i environments/<environment> openwhisk.yml
 ```
 You need to run "initdb" on cloudant **only once** per cloudant database to initialize the db.
 
-**Hint:** In either case the `initdb` mode will ask for your confirmation to drop the database. You can skip this
-confirmation step by adding `-e prompt_user=false` to the command:
+**Hint:** In either case the `initdb` mode will ask for your confirmation to drop the database. To move on you need to enter `CTRL-C + C`. To abort you need to enter `CTRL-C + A`. You can skip this confirmation step by adding `-e prompt_user=false` to the command:
 
 ```
 ansible-playbook -i environments/<environment> couchdb.yml -e mode=initdb -e prompt_user=false
 ```
 
+### Hot-swapping a Single Component
+The playbook structure allows you to clean, deploy or re-deploy a single component as well as the entire Openwhisk stack. Let's assume you have deployed the entire stack using the "openwhisk.yml" playbook. You then make a change to a single component, for example the invoker. You will probably want a new tag on the invoker image so you first build it using:
+
+```
+cd <openwhisk_home>
+gradle :core:dispatcher:distDocker -PdockerImageTag=myNewInvoker
+```
+Then all you need to do is re-deploy the invoker using the new image:
+
+```
+cd ansible
+ansible-playbook -i environments/<environment> invoker.yml
+```
+
+### Cleaning a Single Component
+You can remove a single component just as you would remove the entire deployment stack.
+For example, if you wanted to remove only the controller you would run:
+
+```
+cd ansible
+ansible-playbook -i environments/<environment> controller.yml -e mode=clean
+```
+
+**Caveat:** In distributed environments some components (e.g. Consul, Invoker, etc.) exist on multiple machines. So if you run a playbook to clean or deploy those components, it will affect **all** of the hosts targeted by the component's playbook.
```

Comment:
exactly. I wanted to stress that "component" does not necessarily mean "single host". if you feel it is overhead we can remove the remark.